packages/flutter/lib/src/services/text_input.dart

```diff
@@ -134,20 +135,201 @@ class TextInputType {
+  /// button that represents moving backward, e.g., a left-facing arrow button.
+  ///
+  /// iOS: iOS does not have a keyboard return type of "previous." It is
+  /// inappropriate to choose this [TextInputAction] when running on iOS.
+  previous,
+
+  /// Logical meaning: In iOS apps, it is common for a "Back" button and
+  /// "Continue" button to appear at the top of the screen. However, when the
+  /// keyboard is open, these buttons are often hidden off-screen. Therefore,
+  /// the purpose of the "Continue" return key on iOS is to make the "Continue"
+  /// button available when the user is entering text.
+  ///
+  /// Historical context aside, [TextInputAction.continueAction] can be used any
+  /// time that the term "Continue" seems most appropriate for the given action.
+  ///
+  /// Android: Android does not have an IME input type of "continue." It is
+  /// inappropriate to choose this [TextInputAction] when running on Android.
+  ///
+  /// iOS: Corresponds to iOS's "UIReturnKeyContinue". The title displayed in the
+  /// action button is "Continue". This action is only available on iOS 9.0+.
+  ///
+  /// Note: the reason that this value has "Action" post-fixed to it is because
+  /// "continue" is a reserved word in Dart, as well as many other languages.
+  continueAction,
+
+  /// Logical meaning: The user wants to join something, e.g., a wireless
+  /// network.
+  ///
+  /// Android: Android does not have an IME input type of "join." It is
+  /// inappropriate to choose this [TextInputAction] when running on Android.
+  ///
+  /// iOS: Corresponds to iOS's "UIReturnKeyJoin". The title displayed in the
+  /// action button is "Join".
+  join,
+
+  /// Logical meaning: The user wants routing options, e.g., driving directions.
+  ///
+  /// Android: Android does not have an IME input type of "route." It is
+  /// inappropriate to choose this [TextInputAction] when running on Android.
+  ///
+  /// iOS: Corresponds to iOS's "UIReturnKeyRoute". The title displayed in the
+  /// action button is "Route".
+  route,
+
+  /// Logical meaning: Initiate a call to emergency services.
+  ///
+  /// Android: Android does not have an IME input type of "emergencyCall." It is
+  /// inappropriate to choose this [TextInputAction] when running on Android.
+  ///
+  /// iOS: Corresponds to iOS's "UIReturnKeyEmergencyCall". The title displayed
+  /// in the action button is "Emergency Call".
+  emergencyCall,
+
+  /// Logical meaning: Insert a newline character in the focused text input,
+  /// e.g., [TextField].
+  ///
+  /// Android: Corresponds to Android's "IME_ACTION_NONE". The OS displays a
+  /// button that represents a new line, e.g., a carriage return button.
+  ///
+  /// iOS: Corresponds to iOS's "UIReturnKeyDefault". The title displayed in the
+  /// action button is "return".
+  ///
+  /// Notice that [TextInputAction.newline] is a term that exists in Flutter
```

Comment:
rather than "notice that .. is a term that exists ..", consider just "The term .. exists .."